diff --git a/lib/matplotlib/tests/test_rcparams.py b/lib/matplotlib/tests/test_rcparams.py
--- a/lib/matplotlib/tests/test_rcparams.py
+++ b/lib/matplotlib/tests/test_rcparams.py
@@ -546,3 +546,8 @@ def test_deprecation(monkeypatch):
         mpl.rcParams["svg.hashsalt"] = "foobar"
     assert mpl.rcParams["svg.hashsalt"] == "foobar"  # Doesn't warn.
     mpl.rcParams["svg.hashsalt"] = None  # Doesn't warn.
+
+    mpl.rcParams.update(mpl.rcParams.copy())  # Doesn't warn.
+    # Note that the warning suppression actually arises from the
+    # iteration over the updater rcParams being protected by
+    # suppress_matplotlib_deprecation_warning, rather than any explicit check.
